src/pybind/mgr/nfs/cluster.py

```diff
@@ -203,6 +203,8 @@ def _show_nfs_cluster_info(self, cluster_id: str) -> Dict[str, Any]:
 
     def show_nfs_cluster_info(self, cluster_id: Optional[str] = None) -> Tuple[int, str, str]:
         try:
+            if cluster_id not in available_clusters(self.mgr):
```

Comment:
When `cluster_id` is unset (default: None), this will _always_ evaluate to true, isn't it? Should this be

```
if cluster_id and cluster_id not in ...
```

Reply:
right, so with current code, on running `ceph nfs cluster info` would say `Error ENOENT: cluster does not exist` which is not expected. Thanks for pointing out, will fix it.

Reply:
Done!

Reply:
Do we cover this in one of the tests?

Reply:
nope, will add it.

Reply:
@vshankar `ceph nfs cluster info` works as expected irrespective of this patch. Do we need to test this?

Reply:
That's because `None` is not in the available cluster list. I'll add that change and test anyway.

Reply:
> That's because None is not in the available cluster list. I'll add that change and test anyway.

Is there a related tracker for this?

Reply:
Why does it need a tracker? Is not a bug as you mentioned, just that it a logical check before validating the item in the list.